packages/multicont/multicont.1.0.0-rc.1/opam

```diff
@@ -0,0 +1,36 @@
+opam-version: "2.0"
+maintainer: "Daniel Hillerström <daniel.hillerstrom@ed.ac.uk>"
+authors: "Daniel Hillerström <daniel.hillerstrom@ed.ac.uk>"
+synopsis: "Multi-shot continuations in OCaml"
+description: "This library provides facilities for programming with multi-shot continuations in OCaml"
+homepage: "https://github.com/dhil/ocaml-multicont"
+dev-repo: "git+https://github.com/dhil/ocaml-multicont.git"
+bug-reports: "https://github.com/dhil/ocaml-multicont"
+license: "MIT"
+version: "1.0.0-rc.1"
+
+
+build: [
+  [ make "all" ] {dev}
+  [ make "all" ]
+]
+
+install: [
+  [ make "install" ]
+  [ make "VERSION=%{version}%" "dune-package" ]
```

Comment:
Why you want the `dune-project` file if it is compiled and installed via a Makefile?
Also, you could generate a `multicont.install` file and `opam` would automatically use that to make the installation. In this case, you could statically drop the file in the repository

Reply:
I see. But why don’t just use dune to build all the project then?

Reply:
I was fighting a bit with dune initially, as I need to compile the library differently for byte and native modes. The culprit is how to compile the `multicont_stubs.c` and `fiber_primitives.c` with the correct flags.

Reply:
I’m not sure why you need to create a `dune-package` file though. Dune already understands META file so the dune-package file isn’t necessary and puts more work for you.

Also installing a `dune-package` file would require to add a conflict with `dune < 3.0`. But yeah i don’t think it’s worth having it anyway.

Reply:
> I was fighting a bit with dune initially, as I need to compile the library differently for byte and native modes. The culprit is how to compile the `multicont_stubs.c` and `fiber_primitives.c` with the correct flags.

On the other topic: if you just need to pass different arguments to ocamlc and ocamlopt (If I’m reading your makefile correctly) using `ocamlc_flags`/`ocamlopt_flags` should be enough. See for example:

https://github.com/ocaml/dune/blob/d75bd885262b98075299e9145a2aa3bd1a54a35d/test/blackbox-tests/test-cases/env/envs-and-contexts.t/run.t

Reply:
> I’m not sure why you need to create a `dune-package` file though. Dune already understands META file so the dune-package file isn’t necessary and puts more work for you.
> 
> Also installing a `dune-package` file would require to add a conflict with `dune < 3.0`. But yeah i don’t think it’s worth having it anyway.

I built the dune-package file because dune wouldn't load/link the library correctly otherwise (when I compiled the examples). It should be noted that this library tinkers with the runtime primitives of OCaml. It is also very possible that I failed to get it to work properly before due to my lack of knowledge of dune. I can give it another try at building the project with dune.

Reply:
I have rediscovered the problem that led me away from dune. The problem is that I need to generate different foreign archives for byte and native code. It is not clear to me how I can instruct dune to build, e.g. "libmulticont.a" to link with code produced by ocamlc, and "libmulticontopt.a" to link with code produced by ocamlopt.